ts/src/base/Exchange.ts

```diff
@@ -1958,32 +1958,26 @@ export default class Exchange {
         }
         const market = this.markets[symbol];
         const feeSide = this.safeString (market, 'feeSide', 'quote');
-        let key = 'quote';
-        let cost = undefined;
-        const amountString = this.numberToString (amount);
-        const priceString = this.numberToString (price);
-        if (feeSide === 'quote') {
+        let useQuote = undefined;
+        if (feeSide === 'get') {
             // the fee is always in quote currency
-            cost = Precise.stringMul (amountString, priceString);
-        } else if (feeSide === 'base') {
-            // the fee is always in base currency
-            cost = amountString;
-        } else if (feeSide === 'get') {
-            // the fee is always in the currency you get
-            cost = amountString;
-            if (side === 'sell') {
-                cost = Precise.stringMul (cost, priceString);
-            } else {
-                key = 'base';
-            }
+            useQuote = side === 'sell';
         } else if (feeSide === 'give') {
             // the fee is always in the currency you give
-            cost = amountString;
-            if (side === 'buy') {
-                cost = Precise.stringMul (cost, priceString);
-            } else {
-                key = 'base';
-            }
+            useQuote = side === 'buy';
+        } else {
+            // the fee is always in feeSide currency
+            useQuote = feeSide === 'quote';
+        }
+        let cost = this.numberToString (amount);
+        let key = undefined;
+        if (useQuote) {
+            const priceString = this.numberToString (price);
+            cost = Precise.stringMul (cost, priceString);
+            key = 'quote';
+        } else {
+            cost = amount;
```

Comment:
I don't think we need this because we already assign it on line 1972 and second amount here is a number which breaks the string math on line 1991